tests/CatalogTests/Registration/RegistrationMakerCatalogItemTests.cs

```diff
@@ -81,5 +82,99 @@ private static string GetContentString(StorageContent content)
+            {
+                _graph = new Graph();
+                _graph.Assert(
+                    _graph.CreateUriNode(_catalogUri),
+                    _graph.CreateUriNode(Schema.Predicates.Id),
+                    _graph.CreateLiteralNode("MyPackage"));
+                _graph.Assert(
+                    _graph.CreateUriNode(_catalogUri),
+                    _graph.CreateUriNode(Schema.Predicates.Version),
+                    _graph.CreateLiteralNode("01.02.03+ABC"));
+                _graph.Assert(
+                    _graph.CreateUriNode(_catalogUri),
+                    _graph.CreateUriNode(Schema.Predicates.Published),
+                    _graph.CreateLiteralNode("2015-01-01T00:00:00+00:00"));
+            }
+
+            [Theory]
+            [InlineData(false, false, false, "", "", "http://gallery.org")]
+            [InlineData(true, false, false, "https://test.org", "", "http://gallery.org")]
+            [InlineData(true, true, false, "http://gallery.org/packages/MyPackage/1.2.3/license", "MIT", "http://gallery.org")]
+            [InlineData(true, false, true, "http://gallery.org/packages/MyPackage/1.2.3/license", "", "http://gallery.org")]
+            [InlineData(false, true, false, "http://gallery.org/packages/MyPackage/1.2.3/license", "MIT", "http://gallery.org")]
+            [InlineData(false, false, true, "http://gallery.org/packages/MyPackage/1.2.3/license", "", "http://gallery.org")]
+            [InlineData(false, false, true, "http://gallery.org/packages/MyPackage/1.2.3/license", "", "http://gallery.org/")]
+            [InlineData(false, false, true, "http://gallery.org/packages/MyPackage/1.2.3/license", "", "http://gallery.org//")]
+            public void CreatePageContent_SetsLicenseUrlAndExpressionProperly(bool hasLicenseUrl, bool hasLicenseExpression, bool hasLicenseFile,
+                                                                             string expectedLicenseUrlValue, string expectedLicenseExpressionValue, string galleryBaseAddress)
+            {
+                // Arrange
+                if (hasLicenseUrl)
+                {
+                    _graph.Assert(
+                    _graph.CreateUriNode(_catalogUri),
+                    _graph.CreateUriNode(Schema.Predicates.LicenseUrl),
+                    _graph.CreateLiteralNode("https://test.org"));
+                }
+                if (hasLicenseExpression)
+                {
+                    _graph.Assert(
+                    _graph.CreateUriNode(_catalogUri),
+                    _graph.CreateUriNode(Schema.Predicates.LicenseExpression),
+                    _graph.CreateLiteralNode("MIT"));
+                }
+                if (hasLicenseFile)
+                {
+                    _graph.Assert(
+                    _graph.CreateUriNode(_catalogUri),
+                    _graph.CreateUriNode(Schema.Predicates.LicenseFile),
+                    _graph.CreateLiteralNode("license"));
+                }
```

Comment:
We need to test that file name has no bearing on the output (all kinds of extensions, including invalid ones, paths with subdirectories, too).  Probably separate test, otherwise it would explode this one's input.